src/main/java/org/openstreetmap/atlas/checks/validation/areas/OverlappingAOIPolygonCheck.java

```diff
@@ -0,0 +1,237 @@
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Checks for areas, that are not flagged, and pass any of the TaggableFilters in
+        // aoiFilters. aoiFiltersTest() and aoiFilters is used in place of a single TaggableFilter
+        // so that each filter may be tested separately later.
+        return object instanceof Area && !this.isFlagged(object.getIdentifier())
+                && aoiFiltersTest(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+
+        final Area aoi = (Area) object;
+        final Polygon aoiPolygon = aoi.asPolygon();
+        boolean hasOverlap = false;
+
+        // Set of overlapping area AOIs
+        final Set<Area> overlappingArea = Iterables.stream(object.getAtlas().areasIntersecting(
+                aoiPolygon.bounds(),
+                area -> area.getIdentifier() != aoi.getIdentifier() && area.intersects(aoiPolygon)
+                        && aoiFiltersTest(area) && !this.isFlagged(area.getIdentifier())))
+                .collectToSet();
+
+        final CheckFlag flag = new CheckFlag(this.getTaskIdentifier(object));
+        flag.addObject(object);
+
+        // Test each overlapping AOI to see if it overlaps enough and passes the same AOI filter as
+        // the object
+        for (final Area area : overlappingArea)
+        {
+            if (this.hasMinimumOverlapProportion(aoi.asPolygon(), area.asPolygon()))
+            {
+                if (aoiFiltersTest(object, area))
+                {
+                    flag.addObject(area);
+                    flag.addInstruction(this.getLocalizedInstruction(0, object.getOsmIdentifier(),
+                            area.getOsmIdentifier()));
+                    this.markAsFlagged(area.getIdentifier());
+                    hasOverlap = true;
+                }
+            }
+        }
+
+        if (hasOverlap)
+        {
+            this.markAsFlagged(object.getIdentifier());
+            return Optional.of(flag);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Uses code from findIntersectionType function in BuildingRoadIntersection
```

Comment:
If there is enough reuse, can you pull out the common functionality into an intersection utility or have a base class for all intersection checks?

Reply:
I am looking into this.

Reply:
I have moved this to a new utility class. Once this PR is merged I will look into updating IntersectingBuildingsCheck to use that class as well. 

IntersectingBuildingsCheck was were the code was originally from. The java doc in this check incorrectly listed it as from BuildingRoadIntersection.